src/tokio.rs

```diff
@@ -40,7 +41,8 @@ impl TokioCharDevice {
     pub async fn new<Filelike: IntoFilelike + AsyncRead + AsyncWrite>(
         filelike: Filelike,
     ) -> io::Result<Self> {
-        Self::_new(File::from_into_filelike(filelike)).await
+        let std_file = std::fs::File::from(filelike.into_filelike());
```

Comment:
this seems possibly problematic to me btw, but i wasn't sure how to get from_filelike working.  just wanna flag it.  same thing is in the new_unchecked function.

Reply:
This looks fine. `into_filelike` needs to be told a type to translate into, and `File` is just a trivial wrapper around an `OwnedFd`.